nmap.cc

```diff
@@ -2060,7 +2071,7 @@ int nmap_main(int argc, char *argv[]) {
           o.numhosts_up--;
           break;
         }
-        o.decoys[o.decoyturn] = currenths->v4source();
+        o.decoys[o.decoyturn] = currenths->source();
```

Comment:
Here I can see only two choices in order to use valid datatype, either make a new function which has return type of `sockaddr_storage` solely for this use or modify this function's type and make changes wherever it is used.